tests/conftest.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+import json
+import apypie
+import requests
+
+from collections import namedtuple
+
+User = namedtuple('User', ['username', 'password', 'name'])
+
+
+def pytest_generate_tests(metafunc):
+    with open(metafunc.config.option.variables[0]) as f:
+        variables = json.load(f)
+    assert variables
+
+    if 'url' in metafunc.fixturenames:
+        base_url = f"https://{variables['server_hostname']}"
```

Comment:
This should use the [existing base-url](https://pypi.org/project/pytest-base-url/) that we pass.

Reply:
I'm effectively dropping the use of `--base-url` in favor of configuring all dynamic inputs through `variables.json` rather than splitting them across two methods of specification.

Reply:
And I disagree with that. Mostly because you break the existing tests. pytest-selenium depends on pytest-base-url and uses it.